app/models/customer.rb

```diff
@@ -0,0 +1,11 @@
+class Customer < ApplicationRecord
+  has_many :customers_items
+  has_many :items, through: :customers_items
+
+  def self.find_supermarket_shoppers(supermarket)
```

Comment:
This looks good! I will leave a comment in your supermarket model that will show you an easier solution for AR but your SQL is solid (and it's always okay to use `find_by_sql` instead of AR IMO to complete your query as long as you attempt both!

Reply:
Realizing your Supermarket model isn't coming up bc there are no changes so here's the suggestion:

Something that would make the joining here a lot easier to find unique customers: set up the Supermarket model to take advantage of the associations that items, its child object has. 
Similar to how you use it with many-to-many relationships, you could add the following association to the Supermarket model:
`has_many :customers, through: :items`

You're telling AR: Items has an association to customer, Items belongs to this model, so create a direct associative link from this model all the way to customers. Under the hood, AR will build an association that gets all the customers associated with the items that belong to this store. This is a multi-step join under the hood (like what you wrote here), but with AR, we can just call `supermarket.customers` and get the results!